redisinsight/ui/src/pages/vector-search/query/useQuery.ts

```diff
@@ -0,0 +1,378 @@
+  })
+
+const useQuery = () => {
+  const { instanceId } = useParams<{ instanceId: string }>()
+  const scrollDivRef = useRef<HTMLDivElement>(null)
+
+  const [query, setQuery] = useState('')
+  const [items, setItems] = useState<CommandExecutionUI[]>([])
+  const [clearing, setClearing] = useState(false)
+  const [processing, setProcessing] = useState(false)
+  const [isLoaded, setIsLoaded] = useState(false)
+
+  const resultsMode = ResultsMode.Default
+  const activeRunQueryMode = RunQueryMode.ASCII
+
+  useEffect(() => {
+    const loadHistory = async () => {
+      try {
+        const commandsHistory = await getLocalWbHistory(instanceId)
+        if (Array.isArray(commandsHistory)) {
+          const sortedHistory = commandsHistory.map((item) => ({
+            ...item,
+            command: item.command || EMPTY_COMMAND,
+            emptyCommand: !item.command,
+          }))
+          setItems(sortCommandsByDate(sortedHistory))
+        }
+        setIsLoaded(true)
+      } catch (error) {
+        setIsLoaded(true)
+      }
+    }
+
+    loadHistory()
+  }, [instanceId])
+
+  const prepareNewItems = (
+    commands: string[],
+    commandId: string,
+  ): CommandExecutionUI[] =>
+    commands.map((command, i) => ({
+      command,
+      id: commandId + i,
+      loading: true,
+      isOpen: true,
+      error: '',
+    }))
+
+  const insertNewItems = (
+    newItems: CommandExecutionUI[],
+    isGroup: boolean,
+    commandId: string,
+  ) => {
+    setItems((prevItems) => {
+      let updatedItems = isGroup
+        ? [
+            {
+              command: `${newItems.length} - Command(s)`,
+              id: commandId,
+              loading: true,
+              isOpen: true,
+              error: '',
+            },
+            ...prevItems,
+          ]
+        : [...newItems, ...prevItems]
+
+      if (updatedItems.length > WORKBENCH_HISTORY_MAX_LENGTH) {
+        updatedItems = updatedItems.slice(0, WORKBENCH_HISTORY_MAX_LENGTH)
+      }
+
+      return updatedItems
+    })
+  }
+
+  const handleApiSuccess = async (
+    data: CommandExecution[],
+    commandId: string,
+    restCommands: string[][],
+    executeParams: CodeButtonParams,
+    _isGroup: boolean,
+    isNewCommand: boolean,
+  ) => {
+    setItems((prevItems) => {
+      const updatedItems = prevItems.map((item) => {
+        const result = data.find((_, i) => item.id === commandId + i)
+        if (result) {
+          return {
+            ...result,
+            loading: false,
+            error: '',
+            isOpen: !isSilentMode(resultsMode),
+          }
+        }
+        return item
+      })
+
+      return sortCommandsByDate(updatedItems)
+    })
+
+    await addCommands(reverse(data))
+
+    if (restCommands.length > 0) {
+      const nextCommands = restCommands[0]
+      if (nextCommands?.length) {
+        setTimeout(() => {
+          handleSubmit(nextCommands.join('\n'), undefined, executeParams)
+        }, 100)
```

Comment:
It looks like a fake wait, do you know why we need it, what we're trying to bypass here?

Reply:
Yes, it is a leftover. My first implementation was in memory, without sending it to the API, so I forgot to remove it. I spent some time checking if we need to do it anyways, but for some reasonable size of commands, without it, it works just as fine.